Bottom: da003840aa5b76224d070c7fae566668e12dab38
Top:    58468d0687a55cdf047c8dede587a62478b19f9f
Author: = <=>
Date:   2016-12-08 17:47:06 +0800

macvlan: use skb_direct_xmit()

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/net/macvtap.c b/drivers/net/macvtap.c
index 52a9d81..98e1374 100644
--- a/drivers/net/macvtap.c
+++ b/drivers/net/macvtap.c
@@ -780,8 +780,8 @@ static ssize_t macvtap_get_user(struct macvtap_queue *q, struct msghdr *m,
 	}
 
 	if (vlan) {
-		skb->dev = vlan->dev;
-		dev_queue_xmit(skb);
+		skb->dev = vlan->lowerdev;
+		skb_direct_xmit(skb, false);
 	} else {
 		kfree_skb(skb);
 	}
